microsoft/MixedRealityToolkit-Unity
# Button Configuration Helper

## Overview
- Adds a script to our standard button assets which gathers all commonly modified elements into one place:

![ButtonRecording1](https://user-images.githubusercontent.com/9789716/70167111-e3175600-167a-11ea-9c52-444509c06105.gif)

- Buttons can be modified at runtime:

![ButtonRecording2](https://user-images.githubusercontent.com/9789716/70167525-c92a4300-167b-11ea-8558-067243257617.gif)

- Adds new methods for icon rendering to standard buttons (sprite and character, in addition to our current quad technique). Icons can be configured in the ButtonConfigHelper script:

![ButtonConfigHelper](https://user-images.githubusercontent.com/9789716/67803606-a4262d00-fa4a-11e9-85cd-07752035c349.PNG)
- Introduces ButtonIconSet asset, which can store assets for rendering all three icon styles:

![ButtonConfigHelper2](https://user-images.githubusercontent.com/9789716/67803619-aa1c0e00-fa4a-11e9-86ad-439588cb53db.PNG)

- Makes it easier to change icons at runtime with helper methods.

## Font Asset
Font icons were tested using the MDL2 font asset. Because we can't publicly distribute this asset, you will have to [download it here.](https://aka.ms/SegoeFonts?ocid=cxfluent-v1Oct-fluent-segoe)
Microsoft Design:
https://www.microsoft.com/design/fluent/#/windows
HoloLens Icon Font:
https://aka.ms/hololensiconfont

## Changes
- Fixes: #6207

## Verification
- Verify that there are no unintended interactions with Intractable scripts.
- Verify that this is a non-breaking change, and that projects with modified icon assets will not be affected.
- Write tests.

> As a reviewer, it is possible to check out this change locally by using the following
> commands (substituting {PR_ID} with the ID of this pull request):
>
> git fetch origin pull/{PR_ID}/head:name_of_local_branch
>
> git checkout name_of_local_branch
